File: problems_051to075/problem_063/solution.py
import os
import sys
from math import ceil

PROJECT_PATH = os.path.dirname(os.path.dirname(os.path.dirname(os.path.abspath(__file__))))
sys.path.append(PROJECT_PATH)

from measurement import Stopwatch


class Solution(Stopwatch):
    def __init__(self):
        super().__init__()

    def execute(self):
        answer = 0

        num = 0
        while True:
            num += 1

            min_base = ceil(10 ** ((num-1) / num))
            max_base = 10

            base_count = max_base - min_base

            if base_count == 0:
                break

            answer += base_count

        return answer

if __name__ == '__main__':
    soln = Solution()
    result = soln.execute()
    print(result)

File: problems_051to075/problem_063/solution.ts
import { SolutionFunction } from '../../lib/types';

const solution: SolutionFunction = () => {
  let answer: ReturnType<SolutionFunction> = 0;

  let num = 0;
  const maxBase = 10;
  let minBase: number | null = null;
  let baseCount: number | null = null;

  while (baseCount !== 0) {
    if (baseCount !== null) {
      answer += baseCount;
    }

    num += 1;
    minBase = Math.ceil(10 ** ((num - 1) / num));
    baseCount = maxBase - minBase;
  }

  return answer;
};

export default solution;
